app/src/main/java/org/wikipedia/descriptions/DescriptionEditFragment.kt

```diff
@@ -0,0 +1,448 @@
+package org.wikipedia.descriptions
+
+import android.app.Activity
+import android.content.ActivityNotFoundException
+import android.content.Intent
+import android.os.Bundle
+import android.os.Handler
+import android.speech.RecognizerIntent
+import android.text.TextUtils
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.apache.commons.lang3.StringUtils
+import org.wikipedia.Constants
+import org.wikipedia.Constants.InvokeSource
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.activity.FragmentUtil
+import org.wikipedia.analytics.DescriptionEditFunnel
+import org.wikipedia.analytics.SuggestedEditsFunnel
+import org.wikipedia.auth.AccountUtil.isLoggedIn
+import org.wikipedia.csrf.CsrfTokenClient
+import org.wikipedia.databinding.FragmentDescriptionEditBinding
+import org.wikipedia.dataclient.Service
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwException
+import org.wikipedia.dataclient.mwapi.MwQueryResponse
+import org.wikipedia.dataclient.page.PageSummary
+import org.wikipedia.dataclient.wikidata.EntityPostResponse
+import org.wikipedia.edit.Edit
+import org.wikipedia.json.GsonUnmarshaller
+import org.wikipedia.language.AppLanguageLookUpTable
+import org.wikipedia.login.LoginClient.LoginFailedException
+import org.wikipedia.page.PageTitle
+import org.wikipedia.settings.Prefs
+import org.wikipedia.suggestededits.PageSummaryForEdit
+import org.wikipedia.suggestededits.SuggestedEditsSurvey
+import org.wikipedia.suggestededits.SuggestionsActivity
+import org.wikipedia.util.DeviceUtil
+import org.wikipedia.util.FeedbackUtil
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.log.L
+import java.io.IOException
+import java.util.*
+import java.util.concurrent.TimeUnit
+import java.util.regex.Pattern
+
+class DescriptionEditFragment : Fragment() {
+    interface Callback {
+        fun onDescriptionEditSuccess()
+        fun onBottomBarContainerClicked(action: DescriptionEditActivity.Action)
+    }
+
+    private var _binding: FragmentDescriptionEditBinding? = null
+     val binding get() = _binding!!
+    private var pageTitle: PageTitle? = null
+    private var sourceSummary: PageSummaryForEdit? = null
+    private var targetSummary: PageSummaryForEdit? = null
+    private var highlightText: String? = null
+    private var csrfClient: CsrfTokenClient? = null
+    private lateinit var funnel: DescriptionEditFunnel
+    private lateinit var action: DescriptionEditActivity.Action
+    private lateinit var invokeSource: InvokeSource
```

Comment:
Please make sure to put `lateinit var` to the top of these values for the reading experience.